examples/with-next/app/layout.tsx

```diff
@@ -1,6 +1,6 @@
 import type { Metadata } from 'next'
 import { Inter } from 'next/font/google'
-import { MockProvider } from './mockProvider'
+import { MSWProvider } from './msw-provider'
```

Comment:
While I think it a noble idea to collocate the setup within the same component, it yields more files to create for the end user. 

Your initial setup was better, in my opinion.

- server setup inline in `layout.tsx`;
- `MockProvider` as a single copy-pastable file you import in `layout.tsx` and wrap your children in it.

Reply:
I really like how you utilize `layout.tsx` as both client-side and server-side entrypoints!